### ISSUE TITLE
Details about the BrainWeb data that is being used

### ISSUE THREAD

**Issue by srivathsapv** [StefanDenn3r/Unsupervised_Anomaly_Detection_Brain_MRI#2]
Can you explain in detail about how you download the BrainWeb dataset? The website does not have much information and I am able to download only one series at a time. How do you download the tissue labels and skull map? How many patients/scans are you using from Brain Web?

**Comment by amirmmi** [comment]
@StefanDenn3r hi and thanks for sharing this code,
I have the same problem with @srivathsapv, please guide us
best regards

**Comment by srivathsapv** [comment]
@amirmmi On a side note, I was able to use [IXI](https://brain-development.org/ixi-dataset/) and [ADNI](http://adni.loni.usc.edu/) datasets successfully.

**Comment by amirmmi** [comment]
@srivathsapv 
Thank you very much for your help

**Comment by StefanDenn3r** [comment]
Hi Guys, 
sorry for the late reply and thanks for opening the issue. 
BrainWeb dataset: Please keep in mind that this is a simulated dataset. This was more meant as an easy introduction to the code base since we can quite easily overfit on it and verify that everything is correctly set up. However, I agree, it is quite tedious to download each series at a time. Maybe it makes sense to change this to another dataset, which is easier to deal with. I'll add it as a todo for me. If _you_ would like to do that, feel free to open a pull request. I'd really appreciate it! 
For actual experiments, I'd recommend to rather use a dataset with real patients, such as the [MS ISBI 2015 dataset](https://smart-stats-tools.org/lesion-challenge-2015). 

@srivathsapv I'd really appreciate if you could make a pull request with the IXI and ADNI dataset since it seems like you have already implemented it and others would benefit from it. That would be great!

**Comment by srivathsapv** [comment]
@StefanDenn3r To make it work with IXI and ADNI, my changes were mostly in `dataloaders/BRAINWEB.py` itself. But yeah, sure I can create a PR soon.

**Comment by nurafiqah78** [comment]
@srivathsapv hi there. I'm having difficulty to use BRAINWEB dataset too. hence, i saw your comment and decided to use IXI dataset. appreciate if you can share your code change in BRAINWEB.py for me to double check it with mine since im unable to run using that dataset

**Comment by engr-zeeshan** [comment]
I am using Brainweb dataset, placed in relevant folders but following error

![Screenshot_5](https://user-images.githubusercontent.com/6541601/123233617-a4165300-d4f3-11eb-9355-f86e89d6b35a.jpg)

**Comment by AndresJGUTP** [comment]
@srivathsapv did you the pull request?

**Comment by Shancong-Mou** [comment]
Hi All, any update? I am facing the same dataset issue.

**Comment by Shancong-Mou** [comment]
> Hi All, any update? I am facing the same dataset issue.

@StefanDenn3r

**Comment by StefanDenn3r** [comment]
Hello, 
unfortunately, I don't like sharing the Brainweb data as a folder, since this data does not belong to us. 
However, I am working on an automatic download of the BrainWeb dataset, which should make it very easy for you to get started working with the code base. Please give me some time for that. I will provide you with an update here in the next few days.

Best regards, 
Stefan

**Comment by StefanDenn3r** [comment]
Hello everybody, 
please have a look at the latest commit. I added a script for downloading the Brainweb dataset. 
Furthermore, I also updated the Jupiter Notebook, which now uses this script to download the dataset and let's you start right away with training your models. 
Please try it out! [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/StefanDenn3r/Unsupervised_Anomaly_Detection_Brain_MRI/blob/master/Unsupervised%20Anomaly%20Detection%20Brain-MRI.ipynb)
For now, I will close this issue. If you have any issues, please feel free to reopen it. 

Best, 
Stefan

**Comment by Shancong-Mou** [comment]
> Hello everybody, please have a look at the latest commit. I added a script for downloading the Brainweb dataset. Furthermore, I also updated the Jupiter Notebook, which now uses this script to download the dataset and let's you start right away with training your models. Please try it out! [![Open In Colab](https://camo.githubusercontent.com/84f0493939e0c4de4e6dbe113251b4bfb5353e57134ffd9fcab6b8714514d4d1/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/StefanDenn3r/Unsupervised_Anomaly_Detection_Brain_MRI/blob/master/Unsupervised%20Anomaly%20Detection%20Brain-MRI.ipynb) For now, I will close this issue. If you have any issues, please feel free to reopen it.
> 
> Best, Stefan

Thanks, stefan, that is awesome! By the way, could you please also post the performance of each method on the Brinaweb dataset? Seems like retraining from scratch takes a lot of time.

**Comment by StefanDenn3r** [comment]
Hi, 
the Brainweb Dataset should rather be considered as an example dataset to give you an idea of how to integrate your own dataset and for validating if the whole pipeline is set up correctly, i.e. if the gradients look reasonable, the losses are decreasing and so on. 
Therefore, there is not really a performance to report. 

Best, 
Stefan
